docs/walkthrough.md

```diff
@@ -0,0 +1,500 @@
+its own namespace called `prom`.  Most of the sample commands and files
+are namespace-agnostic, but there are a few commands that rely on
+namespace.  If you're using a different namespace, simply substitute that
+in for `prom` when it appears.
+
+### Prometheus Configuration ###
+
+If you've never deployed Prometheus before, you'll need an appropriate
+Prometheus configuration.  There's a extensive sample Prometheus
+configuration in the Prometheus repository
+[here](https://github.com/prometheus/prometheus/blob/master/documentation/examples/prometheus-kubernetes.yml).
+Be sure to also read the Prometheus documentation on [configuring
+Prometheus](https://prometheus.io/docs/operating/configuration/).
+
+For the purposes of this walkthrough, you'll need the following
+configuration options to be set:
+
+<details>
+
+<summary>prom-cfg.yaml</summary>
+
+```yaml
+# a short scrape interval means you can respond to changes in
+# metrics more quickly
+global:
+  scrape_interval: 15s
+
+# you need a scrape configuration for scraping from pods
+scrape_configs:
+- job_name: 'kubernetes-pods'
+  # if you want to use metrics on jobs, set the below field to
+  # true to prevent Prometheus from setting the `job` label
+  # automatically.
+  honor_labels: false
+  kubernetes_sd_configs:
+  - role: pod
+  # skip verification so you can do HTTPS to pods
+  tls_config:
+    insecure_skip_verify: true
+  # make sure your labels are in order
+  relabel_configs:
+  # these labels tell Prometheus to automatically attach source
+  # pod and namespace information to each collected sample, so
+  # that they'll be exposed in the custom metrics API automatically.
+  - source_labels: [__meta_kubernetes_namespace]
+    action: replace
+    target_label: namespace
+  - source_labels: [__meta_kubernetes_pod_name]
+    action: replace
+    target_label: pod
+  # these labels tell Prometheus to look for
+  # prometheus.io/{scrape,path,port} annotations to configure
+  # how to scrape
+  - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scrape]
+    action: keep
+    regex: true
+  - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_path]
+    action: replace
+    target_label: __metrics_path__
+    regex: (.+)
+  - source_labels: [__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]
+    action: replace
+    regex: ([^:]+)(?::\d+)?;(\d+)
+    replacement: $1:$2
+    target_label: __address__
+  - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scheme]
+    action: replace
+    target_label: __scheme__
+    regex: (.+)
+```
+
+</details>
+
+Place your full configuration (it might just be the above) in a file (call
+it `prom-cfg.yaml`, for example), and then create a ConfigMap containing
+it:
+
+```shell
+$ kubectl create configmap prometheus --from-file=prometheus.yml=prom-cfg.yaml
+```
+
+You'll be using this later when you launch Prometheus.
+
+### Launching the Deployment ###
```

Comment:
Can we break out configuring Prometheus and deploying it from this section and just say: If you want to go the easy way; run these three commands and use the prom operator. If you want to go the custom way, look at this doc.